From b6d8dea6d8718b270dd0e2bd25f8e4612ee34e62 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 28 Jun 2023 14:44:50 +0800
Subject: [PATCH] fix compile error for gcc 13.1

After gcc is upgraded to 13.1, a compile error:
...
|tensorflow/lite/kernels/internal/spectrogram.cc:46:22: error: 'uint32_t' was not declared in this scope
|   46 | inline int Log2Floor(uint32_t n) {
      |                      ^~~~~~~~
|tensorflow/lite/kernels/internal/spectrogram.cc:20:1: note: 'uint32_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
...

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 tensorflow/lite/kernels/internal/spectrogram.cc | 1 +
 tensorflow/tsl/lib/io/cache.h                   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/tensorflow/lite/kernels/internal/spectrogram.cc b/tensorflow/lite/kernels/internal/spectrogram.cc
index a832962a38d..3ee2d036dd4 100644
--- a/tensorflow/lite/kernels/internal/spectrogram.cc
+++ b/tensorflow/lite/kernels/internal/spectrogram.cc
@@ -17,6 +17,7 @@ limitations under the License.
 
 #include <assert.h>
 #include <math.h>
+#include <cstdint>
 
 #include "third_party/fft2d/fft.h"
 
diff --git a/tensorflow/tsl/lib/io/cache.h b/tensorflow/tsl/lib/io/cache.h
index f894c5916d5..4e509cea05c 100644
--- a/tensorflow/tsl/lib/io/cache.h
+++ b/tensorflow/tsl/lib/io/cache.h
@@ -16,6 +16,7 @@ limitations under the License.
 #ifndef TENSORFLOW_TSL_LIB_IO_CACHE_H_
 #define TENSORFLOW_TSL_LIB_IO_CACHE_H_
 
+#include <cstdint>
 #include "tensorflow/tsl/platform/stringpiece.h"
 
 // A Cache is an interface that maps keys to values.  It has internal
